syntax = "proto3";

import "google/protobuf/empty.proto";
import "google/protobuf/wrappers.proto";

service Node {

    rpc stop (google.protobuf.Empty) returns (google.protobuf.Empty);

    rpc update_sensitivity (UpdateSensitivityRequest) returns (google.protobuf.Empty);

    rpc is_recording (NodeIdParameterRequest) returns (google.protobuf.BoolValue);

    rpc record (SwitchRecordingRequest) returns (google.protobuf.Empty);

    rpc stop_recording (SwitchRecordingRequest) returns (google.protobuf.Empty);

    rpc add_camera (CameraInfo) returns (google.protobuf.Empty);

    rpc remove_camera (CameraIdParameterRequest) returns (google.protobuf.Empty);

    rpc get_snapshot_url (NodeIdParameterRequest) returns (google.protobuf.StringValue);
}

message UpdateSensitivityRequest {
    uint32 node_id = 1;
    double sensitivity = 2;
}

message NodeIdParameterRequest {
    uint32 node_id = 1;
}

message CameraIdParameterRequest {
    uint32 camera_id = 1;
}

message SwitchRecordingRequest {
    repeated uint32 cameras_ids = 1;
}

message CameraInfo {
    uint32 id = 1;
    string name = 2;
    string ip = 3;
    uint32 http_port = 4;
    uint32 streaming_port = 5;
    sint32 frame_width = 6;
    sint32 frame_height = 7;
    sint32 frame_rate = 8;
    string user = 9;
    string password = 10;
    string model = 11;
    CameraConfigurations configurations = 12;
}

message CameraConfigurations {
    double sensitivity = 1;
    bool recording = 2;
}
